### DISCUSSION TITLE
Automate OpenVPN User Creation

### DISCUSSION THREAD

**Discussion by krvelinov** [pfsensible/core#71] [Q&A]
Hi.

Excuse me if this has already been answered already.

I am trying to fully automate my OpenVPN Setup. 
Currently i am able to create a new CA and OpenVPN Server via the modules, however i can't see an option to create an internal certificate when creating a new user in pfsensible.core.pfsense_user.

Is this currently available or am i missing something? 

Thank you in advance :)

**Comment by opoplawski** [comment]
Development of these modules is primarily driven by the needs of the developer(s) who have very limited time.  We don't support creating user certificates yet and unlikely will without help.  PRs are welcome.

**Comment by stefangweichinger** [comment]
As far as I see the original question is obsolete now anyway: I could create certs already yesterday.
But maybe the OP is interested in discussing OpenVPN automation:

Sorry for asking such a meta question here, I know I could/should build my playbooks step by step, but maybe I can avoid reinventing wheels here:

I have to generate a dozen of local users on a pfSense-CE-2.8.1, the users should all have their certs generated and ideally I would love all their certs and keys to be exported to files on my ansible controller for further usage (in other scripts: I want to prepare ovpn-satellites with these users in a second step).

Additionally I need a openvpn-CSO for each user on that pfSense generated ...

The roles and collections here are able to do that, I wonder if there are pointers to playbooks combining them already. This would be very helpful!

thanks for any pointers!

EDIT: I have my first steps tested already, and work on proper structure. Maybe someone likes to discuss and improve that.
